pkg/tso/keyspace_group_manager.go

```diff
@@ -1268,51 +1268,62 @@ func (kgm *KeyspaceGroupManager) mergingChecker(ctx context.Context, mergeTarget
 		if len(mergeMap) > 0 {
 			continue
 		}
+		log.Info("all the keyspace group primaries in the merge list are gone, "+
+			"start to calculate the newly merged TSO",
+			zap.String("member", kgm.tsoServiceID.ServiceAddr),
+			zap.Uint32("merge-target-id", mergeTargetID),
+			zap.Any("merge-list", mergeList))
 		// All the keyspace group primaries in the merge list are gone,
-		// update the newly merged TSO to make sure it is greater than the original ones.
+		// calculate the newly merged TSO to make sure it is greater than the original ones.
 		var mergedTS time.Time
 		for _, id := range mergeList {
 			ts, err := kgm.tsoSvcStorage.LoadTimestamp(am.getKeyspaceGroupTSPath(id))
-			if err != nil || ts == typeutil.ZeroTime {
+			if err != nil {
```

Comment:
Will the merge use the wrong ts if the source has not been initialized?

Reply:
I didn't think of such a case. Do you have any specific concerns?

Reply:
I added a test case for this kind of situation. PTAL.